spharpy/spherical.py

```diff
@@ -1,104 +1,599 @@
+        The default is ``'acn'``.
+        (FuMa is only supported up to 3rd order)
+    inverse_transform : str, optional
+        Inverse transform type, either ``'pseudo_inverse'`` or
+        ``'quadrature'``. The default is None.
+    weights : array-like, optional
+        Sampling weights for the quadrature transform.
+        Required if `quadrature` is chosen.
+    condon_shortley : bool, optional
+        Whether to include the Condon-Shortley phase term.
+        The default is True.
+    """
 
+    def __init__(
+        self,
+        n_max,
+        coordinates,
+        basis_type="real",
+        normalization="n3d",
+        channel_convention="acn",
+        inverse_transform=None,
+        weights=None,
+        condon_shortley=True
+    ):
+        # self.n_max = n_max
+        # self.weights = weights
+        # self.coordinates = coordinates
+        # self.basis_type = basis_type
+        # self.inverse_transform = inverse_transform
+        # self.channel_convention = channel_convention  # ch. ord. conv.
+        # self.normalization = normalization  # gain norm. conv.
+        self._n_max = n_max
+        self._weights = weights
+        self._coordinates = coordinates
+        self._basis_type = basis_type
+        self._inverse_transform = inverse_transform
+        self._channel_convention = channel_convention  # ch. ord. conv.
+        self._normalization = normalization  # gain norm. conv.
+        self._condon_shortley = condon_shortley
+        self._recompute_basis = True
+        self._recompute_basis_gradient = True
+        self._recompute_inverse = True
+        self._recompute_inverse_gradient = True
+
+        self._basis = None
+        (self._basis_gradient_theta,
+         self._basis_gradient_phi) = (None, None)
+        self._basis_inv = None
+        (self._basis_inv_gradient_theta,
+         self._basis_inv_gradient_phi) = (None, None)
+        # Store previous values for comparison
+        self._prev_n_max = self.n_max
+        self._prev_coordinates = self.coordinates
+        self._prev_basis_type = self.basis_type
+        self._prev_inverse_transform = self.inverse_transform
+        self._prev_channel_convention = self.channel_convention
+        self._prev_normalization = self.normalization
+        self._prev_condon_shortley = self.condon_shortley
+
+    # Properties
+    # property for condon_shortley
+    @property
+    def condon_shortley(self):
+        return self._condon_shortley
+
+    @condon_shortley.setter
+    def condon_shortley(self, value):
+        if not isinstance(value, bool):
+            raise TypeError("condon_shortley must be a boolean value")
+        if value != self._condon_shortley:
+            self._recompute_basis = True
+            self._recompute_basis_gradient = True
+            self._recompute_inverse = True
+            self._recompute_inverse_gradient = True
+        self._condon_shortley = value
```

Comment:
As far as I remember the Condon Shortley phase is only used for complex valued SHs